### PR TITLE
Optimize the package access counts queries

### PR DESCRIPTION
**Pull request by dimaryaz**
Avoid doing a cross product if a package hash belongs to many packages.
Calculate counts for each (eventname, bucket, hash, date) first, then do the join.

### CODE REVIEW

File: lambdas/access_counts/index.py [hunk 1]
```diff
@@ -169,24 +169,40 @@ def sql_escape(s):
...
+    FROM (
+        SELECT
+            eventname,
+            t.bucket AS bucket,
+            name,
+            date,
+            sum(count) AS count
+        FROM (
+            SELECT eventname, bucket, substr(key, 17) AS hash, date, count(*) AS count from object_access_log
+            WHERE substr(key, 1, 16) = '.quilt/packages/'
```
Comment (akarve) [inline lambdas/access_counts/index.py:184]: Easier to maintain if you do `MANIFEST_PREFIX = '.quilt/packages/'` above and `len(MANIFEST_PREFIX)`, etc. here.

File: lambdas/access_counts/index.py [hunk 2]
```diff
@@ -169,24 +169,40 @@ def sql_escape(s):
...
+        CAST(map_agg(date, count) AS JSON) AS counts
+    FROM (
+        SELECT
+            eventname,
+            t.bucket AS bucket,
+            name,
+            date,
+            sum(count) AS count
+        FROM (
+            SELECT eventname, bucket, substr(key, 17) AS hash, date, count(*) AS count from object_access_log
```
Comment (akarve) [inline lambdas/access_counts/index.py:183]: Nice. This is the trick I was missing.

File: lambdas/access_counts/index.py [hunk 3]
```diff
@@ -169,24 +169,40 @@ def sql_escape(s):
...
-        hash,
-        CAST(histogram(date) AS JSON) AS counts
-    FROM object_access_log JOIN package_hashes
-    ON object_access_log.bucket = package_hashes.bucket AND key = concat('.quilt/packages/', hash)
-    GROUP BY 4, 3, 2, 1
+        t.hash AS hash,
+        counts
+    FROM (
+        SELECT eventname, bucket, substr(key, 17) AS hash, CAST(histogram(date) AS JSON) AS counts from object_access_log
+        WHERE substr(key, 1, 16) = '.quilt/packages/'
```
Comment (akarve) [inline lambdas/access_counts/index.py:202]: Same comments as above, prefer a constant.

File: lambdas/access_counts/index.py [hunk 4]
```diff
@@ -169,24 +169,40 @@ def sql_escape(s):
...
+        GROUP BY 4, 3, 2, 1
+    )
     GROUP BY 3, 2, 1
 """)
 
 PACKAGE_VERSION_ACCESS_COUNTS = textwrap.dedent("""\
     SELECT
         eventname,
-        package_hashes.bucket AS bucket,
+        t.bucket AS bucket,
```
Comment (kevinemoore) [inline lambdas/access_counts/index.py:196]: Minor nit: replace 't' with something descriptive (e.g., 'access_totals', 'total_counts', ...).

**Pull request merged**